pkg/apis/events/v1beta1/cloudbuildsource_defaults.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+
+	duckv1beta1 "github.com/google/knative-gcp/pkg/apis/duck/v1beta1"
+	metadataClient "github.com/google/knative-gcp/pkg/gclient/metadata"
+
+	"knative.dev/pkg/ptr"
+)
+
+const (
+	DefaultTopic = "cloud-builds"
+)
+
+func (bs *CloudBuildSource) SetDefaults(ctx context.Context) {
+	ctx = apis.WithinParent(ctx, bs.ObjectMeta)
+	bs.Spec.SetDefaults(ctx)
+	duckv1beta1.SetClusterNameAnnotation(&bs.ObjectMeta, metadataClient.NewDefaultMetadataClient())
+	duckv1beta1.SetAutoscalingAnnotationsDefaults(ctx, &bs.ObjectMeta)
+}
+
+func (bss *CloudBuildSourceSpec) SetDefaults(ctx context.Context) {
+	bss.SetPubSubDefaults(ctx)
+	if bss.Topic == nil {
```

Comment:
this should not be part of the spec. Users shouldn't be able to configure something that is actually non-configurable... It will always be the same topic so we can hardcode it in the reconciler when we need to use it..